src/service-api.ts

```diff
@@ -0,0 +1,42 @@
+// global
+import { FastifyPluginAsync } from 'fastify';
+
+// local
+import { SearchService } from './db-service';
+import { TaskManager } from './task-manager';
+
+const plugin: FastifyPluginAsync = async (fastify) => {
+  const {
+    taskRunner: runner
+  } = fastify;
+  const searchService = new SearchService();
+  const taskManager = new TaskManager(searchService);
+
+  // TODO: schema
+
+  const getTaskByRange = (keyword, range, member) => {
+    switch (range) {
+      case 'title':
+        return taskManager.createSearchByTitleTask(member, keyword);
+      case 'tag':
+        return taskManager.createSearchByTagTask(member, keyword);
+      case 'all':
+        return taskManager.createSearchByAllTask(member, keyword);
+    }
+  };
+
+  // search for items with keyword
+  // range: title, tag, all
+  // TODO: search for author
```

Comment:
Do you plan to do it in this PR? We can address it later.

Reply:
I will do this in a separate PR because it works differently from others

Reply:
I actually also implemented the author search